src/controllers/requestController.js

```diff
@@ -292,6 +292,51 @@ class RequestController {
       return response.sendErrorMessage();
     }
   }
+
+  /**
+   * @description This methods helps users edit requests that are still pending
+   * @param  {object} req - The request object
+   * @param  {object} res - The response object
+   * @returns  {object} The response object
+   */
+  static async editRequest(req, res) {
+    const { requestData } = req;
+    const requestId = req.params.id;
+    const userEmail = req.userData.email;
+    const request = await Request.findOne({
+      where: {
+        id: requestId
+      }
+    });
+    const user = await User.findOne({
+      where: {
+        email: userEmail
+      }
+    });
+    if (!request) {
+      return res.status(404).json({
+        status: 404,
+        error: 'Request not found',
+      });
+    }
+    if (request.user_id === user.id) {
+      if (request.status === 'Pending') {
+        await RequestRepository.update({ id: requestId }, requestData);
```

Comment:
@Alexander96779  review this line 324 to 329 to catch errors

Reply:
I am done